src/i2p.cpp

```diff
@@ -217,6 +217,7 @@ bool Session::Connect(const CService& to, Connection& conn, bool& proxy_error)
     // Refuse connecting to arbitrary ports. We don't specify any destination port to the SAM proxy
     // when connecting (SAM 3.1 does not use ports) and it forces/defaults it to I2P_SAM31_PORT.
     if (to.GetPort() != I2P_SAM31_PORT) {
+        Log("Error connecting to %s, connection refused due to arbitrary port %s", to.ToStringAddrPort(), to.GetPort());
```

Comment:
nit, feel free to ignore: usually "connection refused" means that the connection attempt has been actively declined by the destination host (that is, they send us TCP RST packet). Maybe it would be more clear to say "Refusing to connect to %s because its port is not %d", to.ToStringAddrPort(), I2P_SAM31_PORT. E.g. "Refusing to connect to foo.i2p:1234 because its port is not 0".

Reply:
Sounds good, but I'll leave it as is due to previous reviews.